frontend-wasm/src/component/build_ir.rs

```diff
@@ -0,0 +1,536 @@
+            }
+            Ok(())
+        }
+        Export::ModuleStatic(_) => todo!(),
+        Export::ModuleImport(_) => todo!(),
+        Export::Type(_) => todo!(),
+    }
+}
+
+fn build_export_function(
+    component_instance: &ComponentInstance<'_>,
+    name: &String,
+    func: &CoreDef,
+    ty: &TypeFuncIndex,
+    options: &CanonicalOptions,
+    cb: &mut miden_hir::ComponentBuilder<'_>,
+    config: &WasmTranslationConfig,
+) -> WasmResult<()> {
+    assert_empty_canonical_options(options);
+    let func_ident = match func {
+        CoreDef::Export(core_export) => {
+            let parsed_module = component_instance.module(core_export.instance);
+            let module_name = parsed_module.module.name();
+            let module_ident = miden_hir::Ident::with_empty_span(Symbol::intern(module_name));
+            let func_name = match core_export.item {
+                ExportItem::Index(idx) => match idx {
+                    EntityIndex::Function(func_idx) => parsed_module.module.func_name(func_idx),
+                    EntityIndex::Table(_) => todo!(),
+                    EntityIndex::Memory(_) => todo!(),
+                    EntityIndex::Global(_) => todo!(),
+                },
+                ExportItem::Name(_) => todo!(),
+            };
+            let func_ident = miden_hir::FunctionIdent {
+                module: module_ident,
+                function: miden_hir::Ident::with_empty_span(Symbol::intern(func_name)),
+            };
+            func_ident
+        }
+        CoreDef::InstanceFlags(_) => todo!(),
+        CoreDef::Trampoline(_) => todo!(),
+    };
+    let lifted_func_ty = convert_lifted_func_ty(ty, &component_instance.component_types);
+    let export_name = Symbol::intern(name).into();
+    let Some(export_metadata) = config.export_metadata.get(&export_name) else {
+        return Err(WasmError::MissingExportMetadata(format!(
+            "Export metadata for interface function {:?} not found",
+            &export_name,
+        )));
+    };
+    let export = miden_hir::ComponentExport {
+        function: func_ident,
+        function_ty: lifted_func_ty,
+        invoke_method: export_metadata.invoke_method,
+    };
+    cb.add_export(export_name, export);
+    Ok(())
+}
+
+fn convert_lifted_func_ty(
+    ty: &TypeFuncIndex,
+    component_types: &ComponentTypes,
+) -> LiftedFunctionType {
+    let type_func = component_types[*ty].clone();
+    let params_types = component_types[type_func.params].clone().types;
+    let results_types = component_types[type_func.results].clone().types;
+    let params = params_types
+        .into_iter()
+        .map(|ty| interface_type_to_ir(ty, component_types))
+        .collect();
+    let results = results_types
+        .into_iter()
+        .map(|ty| interface_type_to_ir(ty, component_types))
+        .collect();
+    LiftedFunctionType { params, results }
+}
+
+fn assert_empty_canonical_options(options: &CanonicalOptions) {
```

Comment:
Oh, my bad, it's not actually an assertion. The next PRs for basic wallet and note script component translations will definitely use `CanonicalOptions` so it will be handled in some way for sure. It should've been a `todo!` rather than an assertion.